src/main/java/com/acme/dbo/txlog/Facade.java

```diff
@@ -1,11 +1,41 @@
 package com.acme.dbo.txlog;
 
 public class Facade {
+    public static boolean isDecorated;
+
+    private static void decorateLogPrimitive(String message) {
+        if (Facade.isDecorated) {
+            System.out.println("primitive: " + message);
+        } else {
+            System.out.println(message);
```

Comment:
I see code duplication: both _if_ branches use system output. Copy-paste detected.

Reply:
What is considered as no-duplicate?
Should it be 1 method to write data instead of typing System.out.println 2 times?

Reply:
yep )

Reply:
Agreed.